addons/l10n_in_ewaybill_irn/models/l10n_in_ewaybill.py

```diff
@@ -27,6 +27,17 @@ def _compute_is_process_through_irn(self):
                 )
             )
 
+    def _check_state(self):
+        error_message = super()._check_state()
+        if self.account_move_id and self.is_process_through_irn:
+            l10n_in_edi_response_irn = self.account_move_id._get_l10n_in_edi_response_json().get('Irn')
+            if not l10n_in_edi_response_irn:
+                error_message.append(_(
+                "As the related invoice is eligible for e-invoicing, please generate the e-invoice first. "
+                "Only then can the e-way bill be generated."
+            ))
```

Comment:
```suggestion
                ))
```

Reply:
@prep-odoo Could you still adapt this?  It would confuse everyone who reads these lines later.